### ISSUE TITLE
Playwright tests fail when choosing option in select beyond initial viewport

### ISSUE THREAD

**Issue by jabrks** [adobe/react-spectrum#7516]
### Provide a general summary of the issue here

We've encountered an issue after upgrading from v1.4.1 of the library whereby Playwright tests that try to open a select component found below the fold and choose an option will fail

### 🤔 Expected Behavior?

The test should pass successfully

### 😯 Current Behavior

The test times out whilst trying to locate the option

### 🖥️ Steps to Reproduce

https://github.com/jabrks/react-aria-reproduction

There's a GitHub action that runs on the main branch in the repo that demonstrates the test failure, or you can clone the code and run the tests locally yourself with `npm test`

### Version

react-aria-components 1.5.0

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

macOS

**Comment by snowystinger** [comment]
Probably need to scroll to the select before opening it?
Or does playwright handle that?

**Comment by jabrks** [comment]
I believe Playwright handles it. The test was working as expected with RAC v1.4.1 and below so I had presumed a regression somewhere was to blame rather than anything to do with Playwright itself

**Comment by jabrks** [comment]
I have cherry-picked the commits in the release and it seems as though the issue was introduced as a result of the changes to `@react-aria/overlays` in https://github.com/adobe/react-spectrum/pull/7368

**Comment by jabrks** [comment]
I have cherry-picked the commits in the release to try and narrow down the cause and it seems as though the issue was introduced as a result of the changes to `@react-aria/overlays` in https://github.com/adobe/react-spectrum/pull/7368

**Comment by jabrks** [comment]
I have cherry-picked the commits in the release to try and narrow down the cause and it seems as though the issue was introduced as a result of the changes to `@react-aria/overlays` in https://github.com/adobe/react-spectrum/pull/7368?

**Comment by snowystinger** [comment]
Awesome, thank you for doing that. It's super helpful.
Now we'll just need to determine what in there was the culprit, probably best to start with the changes in `@react-aria/overlays`. Theoretically, there should have been no behavioral changes, just strict mode fixes. So, we should be able to undo changes in each file individually to track down the cause.

If you don't have time, don't worry. I'm just outlining what I would do for whoever comes along next.

**Comment by jabrks** [comment]
I've narrowed it down further to [`useOverlayPosition.ts`](https://github.com/adobe/react-spectrum/blob/326f48154e301edab425c8198c5c3af72422462b/packages/%40react-aria/overlays/src/useOverlayPosition.ts)

**Comment by jabrks** [comment]
Changing this `undefined` back to `null` fixes it for me
https://github.com/adobe/react-spectrum/blob/326f48154e301edab425c8198c5c3af72422462b/packages/%40react-aria/overlays/src/usePopover.ts#L101

I am not entirely clear yet as to why that is the case...

**Comment by jabrks** [comment]
I think I can see the problem. There's a check in `useCloseOnScroll` for if `onClose` is `null` which is no longer being met, resulting in this being called where it previously wasn't 
https://github.com/adobe/react-spectrum/blob/326f48154e301edab425c8198c5c3af72422462b/packages/%40react-aria/overlays/src/useCloseOnScroll.ts#L34-L36

**Comment by jabrks** [comment]
I think I can see the problem. There's a condition in `useCloseOnScroll` for if `onClose` is `null` which is no longer being met, resulting in this being called where it previously wasn't 
https://github.com/adobe/react-spectrum/blob/326f48154e301edab425c8198c5c3af72422462b/packages/%40react-aria/overlays/src/useCloseOnScroll.ts#L34-L36

**Comment by snowystinger** [comment]
I think it'd make the most sense to change the types on `useCloseOnScroll` to include `null`. It looks like the different behavior was added on purpose https://github.com/adobe/react-spectrum/commit/e4ab4895584b6f453157a78219b7b1e4a14adc06

Then go back and change `usePopover` to pass null again.

I'm not sure why TS didn't mark it as dead code.

The other option would be to remove that check entirely. It doesn't fail anything to do that, but I don't know what other implications there are, so it'd need to be tested in storybook. Specifically on an iPad according to that commit/PR.

**Comment by krystian-jedrzejowski-zazoon** [comment]
> I believe Playwright handles it. The test was working as expected with RAC v1.4.1 and below so I had presumed a regression somewhere was to blame rather than anything to do with Playwright itself

I have same behaviour. We have upgraded "react-aria-components": "1.4.1" to ver "1.5.0" and playwright tests became flaky. I have noticed that when I have dropdown component open for selection Playwright do the scroll to find an option and it causes the dropdown to close.

**Comment by RobHannay** [comment]
Thanks for fixing this! <3 I assume this will just go out with the next minor version?

**Comment by LFDanLu** [comment]
Yep, that is correct, provided we don't catch anything broken in the behavior in our final test session before release.
